src/service_mongoose_system_stats.erl

```diff
@@ -0,0 +1,228 @@
+-module(service_mongoose_system_stats).
+-author('jan.ciesla@erlang-solutions.com').
+
+-behaviour(gen_server).
+
+-define(BASE_URL, "https://www.google-analytics.com/batch").
+-define(TRACKING_ID, "UA-151671255-1").
+-define(DEFAULT_REPORT_AFTER, 60 * 60 * 1000).
+-define(STAT_TYPE, [mongoose_system_stats]).
+
+-include("mongoose.hrl").
+
+-export([start/1, stop/0]).
+-export([start_link/0,
+         init/1,
+         handle_event/4,
+         handle_continue/2,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2]).
+
+-record(system_stats_state, {
+    client_id = '',
+    reports = [],
+    report_after = ?DEFAULT_REPORT_AFTER,
+    loop_timer_ref
+    }).
+-type system_stats_state() :: #system_stats_state{}.
+
+-record(service_mongoose_system_stats, {key, value}).
+
+-spec start(proplists:proplist()) -> {ok, pid()}.
+start(_Args) ->
+    Spec = {?MODULE, {?MODULE, start_link, []}, temporary, brutal_kill, worker, [?MODULE]},
+    {ok, _} = ejabberd_sup:start_child(Spec).
+
+-spec stop() -> ok.
+stop() ->
+    ejabberd_sup:stop_child(?MODULE).
+
+-spec start_link() -> {ok, pid()}.
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+-spec init([]) -> {ok, system_stats_state(), {continue, do_init}}.
+init(_Args) ->
+    {ok, #system_stats_state{}, {continue, do_init}}.
+
+-spec handle_continue(do_init, system_stats_state()) ->
+    {noreply, system_stats_state()} | {stop, no_client_id, system_stats_state()}.
+handle_continue(do_init, State) ->
+    telemetry:attach(
```

Comment:
As far as I can tell the `telemetry:attach` makes more sens when done when we know the `client_id` is there, right?

Reply:
Moved telemetry:attach inside case statement.